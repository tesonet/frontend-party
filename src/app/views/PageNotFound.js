import React from "react";

class PageNotFound extends React.Component {
  render() {
    return <h1>Page Not Found!</h1>;
  }
}

export default PageNotFound;
