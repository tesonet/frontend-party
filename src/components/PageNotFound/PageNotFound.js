import React from 'react';

// TODO: style
export default function PageNotFound() {
  return (
    <div>
      Page not found! 404
    </div>
  );
}
