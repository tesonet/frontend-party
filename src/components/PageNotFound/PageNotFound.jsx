import React, { Component } from "react";
import ReactCSSTransitionGroup from "react-addons-css-transition-group";

import styles from "./PageNotFound.module";
import logo from "assets/images/testio.svg";

export default class PageNotFound extends Component {
  render() {
    return (
      <div className={styles.PageNotFound} role="main">
        <ReactCSSTransitionGroup
          transitionAppear={true}
          transitionAppearTimeout={600}
          transitionEnterTimeout={600}
          transitionLeaveTimeout={200}
          transitionName={"SlideIn"}
        >
          <p>
            <figure>
              <img src={logo} alt="Testio" width="200" />
            </figure>
            <strong>PAGE NOT FOUND </strong>
            <a onClick={() => this.props.history.goBack()}>return</a>
          </p>
        </ReactCSSTransitionGroup>
      </div>
    );
  }
}
