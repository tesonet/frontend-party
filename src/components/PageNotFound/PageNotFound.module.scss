.PageNotFound {
  background-color: #fff;
  padding: 20px;
  height: 100%;
  display: grid;
  text-align: center;
  & > * {
    justify-self: center;
    align-self: center;
  }
  strong {
    display: block;
    margin: 40px 0;
    font-size: 2em;
  }
  a {
    color: $primary;
    text-decoration: underline;
    cursor: pointer;
  }
}
