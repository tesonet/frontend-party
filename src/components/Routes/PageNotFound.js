import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';

import { ROUTE_LOGIN } from '../../constants/routes';

const PageNotFound = props => (
  <Redirect
    to={{
      pathname: ROUTE_LOGIN,
      state: { from: props.location },
    }}
  />
);

PageNotFound.propTypes = {
  location: PropTypes.object.isRequired,
};

export default PageNotFound;
