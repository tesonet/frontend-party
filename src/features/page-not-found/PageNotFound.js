import React from 'react';
import style from './PageNotFound.scss';

export default () => (
  <main className={style.pageNotFound}>
    <section>
      <span>404</span>
      <p>The requested path could not be found</p>
    </section>
  </main>
);
