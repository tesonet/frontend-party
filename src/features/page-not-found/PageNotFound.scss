.pageNotFound {
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;

  section {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: row;

    @media (max-width: 768px) {
      flex-direction: column;
    }
  }

  span {
    border-bottom: 0;
    border-right: 1px solid #EAEAEA;
    padding: 0 20px 0 0;
    width: auto;
    font-size: 24px;
    font-weight: 500;
    display: block;
    text-align: center;
    height: 100%;
    line-height: 40px;

    @media (max-width: 768px) {
      padding-left: 20px;
      padding-bottom: 10px;
      border-right: 0;
      border-bottom: 1px solid #EAEAEA;
    }
  }

  p {
    height: 100%;
    line-height: 40px;
    font-size: 14px;
    font-weight: 400;
  }

  span + p {
    margin: 0;
    padding-left: 20px;
  }
}
