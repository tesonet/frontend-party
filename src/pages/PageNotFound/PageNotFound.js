import React from 'react';

function PageNotFound({ location }) {
  document.title = `${document.title} - Page not found!`;

  return (
    <h3>
      Page not found at <code>{location.pathname}</code>
    </h3>
  );
}

export default PageNotFound;
