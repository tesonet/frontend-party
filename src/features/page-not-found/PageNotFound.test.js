import React from 'react';
import Enzyme, { render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PageNotFound from './PageNotFound';

Enzyme.configure({ adapter: new Adapter() });

test('PageNotFound renders without crashing', () => {
  shallow(<PageNotFound />);
});

test('PageNotFound renders to correct snapshot', () => {
  const e = render(<PageNotFound />);
  expect(e.html()).toMatchSnapshot();
});
