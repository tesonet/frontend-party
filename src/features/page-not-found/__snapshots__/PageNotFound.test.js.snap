// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound renders to correct snapshot 1`] = `"<section><span>404</span><p>The requested path could not be found</p></section>"`;
